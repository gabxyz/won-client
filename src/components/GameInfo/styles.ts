import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import * as RibbonStyles from 'components/Ribbon/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    position: relative;
    background: ${theme.colors.white};
    padding: ${theme.spacings.xsmall};
    border-radius: ${theme.border.radius};

    ${RibbonStyles.Wrapper} {
      right: -1rem;
      top: 2rem;
      font-size: ${theme.font.sizes.medium};
      padding: 1.5rem;
      border-radius: ${theme.border.radius} 0 0 ${theme.border.radius};

      &:before {
        border-right-width: 1rem;
      }
    }

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.small};

      ${RibbonStyles.Wrapper} {
        position: relative;
        top: 0;
        right: 0.1rem;
        font-size: ${theme.font.sizes.medium};
        border-radius: ${theme.border.radius};

        &:before {
          border: none;
        }
      }
    `}
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    margin-bottom: ${theme.spacings.small};
    margin-top: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      max-width: 77rem;
      margin-top: ${theme.spacings.xxsmall};
    `}
  `}
`

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;

    > button {
      width: 100%;
      margin-bottom: ${theme.spacings.xxsmall};
    }

    ${media.greaterThan('medium')`
      flex-direction: row-reverse;

      > button {
        width: initial;
        margin-bottom: 0;
      }
    `}
  `}
`

export const HeadWrapper = styled.div`
  display: flex;
  justify-content: space-between;

  > h2 {
    max-width: 60%;
  }

  ${media.greaterThan('medium')`
    > h2 {
      max-width: 100%;
    }
  `}
`

export const PriceWrapper = styled.div`
  margin-top: 1rem;
  ${media.greaterThan('medium')`
    display: flex;
    align-items: center;
    gap: 1.5rem;
    margin-bottom: 4rem;
`}
`

export const Promotional = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    text-decoration: line-through;
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    margin-right: 7rem;

    ${media.greaterThan('medium')`
      margin: 0;
    `}
  `}
`
