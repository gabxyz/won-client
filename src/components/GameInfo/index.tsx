import formatPrice from 'utils/format-price'

import Heading from 'components/Heading'
import Ribbon from 'components/Ribbon'
import CartButton from 'components/CartButton'

import * as S from './styles'
import WishlistButton from 'components/WishlistButton'

export type GameInfoProps = {
  id: string
  title: string
  description: string
  price: number
  basePrice?: number
}

const GameInfo = ({
  id,
  title,
  description,
  basePrice,
  price
}: GameInfoProps) => (
  <S.Wrapper>
    <S.HeadWrapper>
      <Heading color="black" lineBottom>
        {title}
      </Heading>
      <S.PriceWrapper>
        {!!basePrice && <S.Promotional>{formatPrice(basePrice)}</S.Promotional>}
        <Ribbon color="secondary">{formatPrice(price)}</Ribbon>
      </S.PriceWrapper>
    </S.HeadWrapper>

    <S.Description>{description}</S.Description>

    <S.ButtonsWrapper>
      {price !== null && <CartButton id={id} size="medium" hasText />}
      <WishlistButton
        id={id}
        hasText
        size="medium"
        minimal={price !== null}
      ></WishlistButton>
    </S.ButtonsWrapper>
  </S.Wrapper>
)

export default GameInfo
