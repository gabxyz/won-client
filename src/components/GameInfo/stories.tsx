import { Story, Meta } from '@storybook/react/types-6-0'
import GameInfo, { GameInfoProps } from '.'

export default {
  title: 'GameInfo',
  component: GameInfo,
  args: {
    title: 'Game Title',
    description: 'Game Description',
    price: '220.00'
  }
} as Meta

export const Default: Story<GameInfoProps> = (args) => <GameInfo {...args} />
