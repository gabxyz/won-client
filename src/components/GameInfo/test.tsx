import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import GameInfo from '.'

const props = {
  title: 'Game Title',
  description: 'Game Description',
  price: 220
}

describe('<GameInfo />', () => {
  it('should render game infos', () => {
    const { container } = renderWithTheme(<GameInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /game title/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/game description/i)).toBeInTheDocument()
    expect(screen.getByText(/\$220\.00/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render basePrice with a line-through when passed', () => {
    renderWithTheme(<GameInfo {...props} basePrice={250} />)

    expect(screen.getByText('$250.00')).toHaveStyle({
      textDecoration: 'line-through'
    })

    expect(screen.getByText('$220.00')).not.toHaveStyle({
      textDecoration: 'line-through'
    })
  })

  it('should render FREE ribbon when price is 0', () => {
    renderWithTheme(<GameInfo {...props} price={0} />)

    const ribbon = screen.getByText(/free/i)

    expect(ribbon).toBeInTheDocument()
  })

  it('should render Coming Soon ribbon when price is null', () => {
    renderWithTheme(<GameInfo {...props} price={null!} />)

    const ribbon = screen.getByText(/coming soon/i)

    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' })
    expect(ribbon).toBeInTheDocument()
  })

  it('should render buttons', () => {
    renderWithTheme(<GameInfo {...props} />)

    expect(
      screen.getByRole('button', { name: /wishlist/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument()
  })
})
