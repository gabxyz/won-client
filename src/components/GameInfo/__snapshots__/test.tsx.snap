// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render game infos 1`] = `
.c2 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c2::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c5 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #3CD3C1;
  height: 3.6rem;
  padding: 0 2.4rem;
  font-size: 1.4rem;
  right: -2rem;
}

.c5::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c5::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c5::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c9 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c8:focus {
  outline: 1px dashed;
}

.c8:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  background: none;
  color: #F231A5;
}

.c10:focus {
  outline: 1px dashed;
}

.c10:hover {
  background: none;
}

.c10 svg {
  width: 1.5rem;
}

.c10 svg + span {
  margin-left: 0.8rem;
}

.c10:hover {
  color: #e20e8d;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  background: #FAFAFA;
  padding: 1.6rem;
  border-radius: 0.4rem;
}

.c0 .c4 {
  right: -1rem;
  top: 2rem;
  font-size: 1.6rem;
  padding: 1.5rem;
  border-radius: 0.4rem 0 0 0.4rem;
}

.c0 .c4:before {
  border-right-width: 1rem;
}

.c6 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
  margin-top: 1.6rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 > button {
  width: 100%;
  margin-bottom: 0.8rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 > h2 {
  max-width: 60%;
}

.c3 {
  margin-top: 1rem;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }

  .c0 .c4 {
    position: relative;
    top: 0;
    right: 0.1rem;
    font-size: 1.6rem;
    border-radius: 0.4rem;
  }

  .c0 .c4:before {
    border: none;
  }
}

@media (min-width:768px) {
  .c6 {
    max-width: 77rem;
    margin-top: 0.8rem;
  }
}

@media (min-width:768px) {
  .c7 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .c7 > button {
    width: initial;
    margin-bottom: 0;
  }
}

@media (min-width:768px) {
  .c1 > h2 {
    max-width: 100%;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 1.5rem;
    margin-bottom: 4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="black"
    >
      Game Title
    </h2>
    <div
      class="c3"
    >
      <div
        class="c4 c5"
        color="secondary"
      >
        $45.00
      </div>
    </div>
  </div>
  <p
    class="c6"
  >
    Game Description
  </p>
  <div
    class="c7"
  >
    <button
      class="c8"
    >
      <svg
        aria-hidden="true"
        aria-label="Add to cart"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c10"
    >
      <svg
        aria-hidden="true"
        aria-label="Add to Wishlist"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span>
        Add to Wishlist
      </span>
    </button>
  </div>
</div>
`
